src/System.Reflection/tests/AssemblyNameTests.cs

```diff
@@ -304,6 +304,159 @@ public void Version_CurrentlyExecutingAssembly()
+            // No build
+            var expectedVersion = new Version(1, 1);
+            yield return new object[] { expectedVersion, "1.1" };
+            yield return new object[] { expectedVersion, $"1.1.{VersionUnspecifiedStr}" };
+            yield return new object[] { expectedVersion, $"1.1.{VersionUnspecifiedStr}.1" };
+
+            // No revision
+            expectedVersion = new Version(1, 1, 1);
+            yield return new object[] { expectedVersion, "1.1.1" };
+            yield return new object[] { expectedVersion, $"1.1.1.{VersionUnspecifiedStr}" };
+
+            // All components
+            yield return new object[] { new Version(1, 1, 1, 1), "1.1.1.1" };
+            // 65535 causes the component to be considered unspecified. That's not very interesting, so using 65534 instead.
+            yield return new object[] { new Version(65534, 65534, 65534, 65534), "65534.65534.65534.65534" };
+        }
+
+        [Theory]
+        [MemberData(nameof(Constructor_String_VersionTest_MemberData))]
+        [ActiveIssue("https://github.com/dotnet/corert/issues/4321", TargetFrameworkMonikers.UapAot)]
+        [SkipOnTargetFramework(
+            TargetFrameworkMonikers.NetFramework,
+            ".NET Core behavior differs from .NET Framework since it does not want to replicate some bugs")]
+        public static void Constructor_String_VersionTest(Version expectedVersion, string versionStr)
+        {
+            Assert.NotNull(expectedVersion);
+
+            Action<AssemblyName> verify =
+                an =>
+                {
+                    if (expectedVersion == null)
+                    {
+                        Assert.Null(an.Version);
+                    }
+                    else
+                    {
+                        Assert.Equal(expectedVersion, an.Version);
+                    }
+                };
+
+            var assemblyNameFromStr = new AssemblyName("a, Version=" + versionStr);
+            verify(assemblyNameFromStr);
+            verify(new AssemblyName(assemblyNameFromStr.FullName));
+
+            var versionFromStr = new Version(versionStr);
+
+            // The Version class has components of size int32, while AssemblyName(string) only allows uint16 values
+            if (versionStr.IndexOf(VersionUnspecifiedStr, StringComparison.Ordinal) == -1)
+            {
+                Assert.Equal(expectedVersion, versionFromStr);
+            }
+
+            assemblyNameFromStr = new AssemblyName("a, Version=" + versionFromStr);
+            verify(assemblyNameFromStr);
+            verify(new AssemblyName(assemblyNameFromStr.FullName));
+
+            assemblyNameFromStr = new AssemblyName() { Name = "a", Version = expectedVersion };
+            verify(assemblyNameFromStr);
+            verify(new AssemblyName(assemblyNameFromStr.FullName));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.UapAot, "Dynamic assembly loads are not supported")]
```

Comment:
This test is doable on uapaot. You'll need to make sure there's a static reference to the test assemblies in the test project so that ILC will merge them into the closure.